docs/content/modeling/modular-models.mdx

```diff
@@ -0,0 +1,190 @@
+
+To solve this, individual types can be extended within other modules to implement the relations needed.
+
+In order to allow this, there are certain requirements for type extension:
+
+- The extended type must exist
+- A single type can only be extended once per file
+- The relations added must not already exist, or be part of another type extension
+
+## Example
+
+In this example we'll look at how an authorization model for a SaaS compny with a issue tracking and wiki software could implement modular models.
+
+### Core
+
+It's likely there will be a core set of types owned by a team that manages the overall identity for the company, this would provide the basics usch as users, organizations and groups that can be used by each respective product area.
+
+```dsl.openfga
+module core
+
+type user
+
+type organization
+  relations
+    define member: [user]
+    define admin: [user]
+
+type group
+  relations
+    define member: [user]
+```
+
+### Issue tracking
+
+In the issue tracking software we'd likely separate out the project and issue related types into separate files, we'll also extend the `organization` type here so that we can add a relation specific to the issue tracking feature, which is the ability to authorize who can create a project.
+
+```dsl.openfga
+module issue-tracker
+
+extend type organization
+  relations
+    define can_create_project: admin
+
+type project
+  relations
+    define organization: [organization]
+    define viewer: member from organization
+```
+
+```dsl.openfga
+module issue-tracker
+
+type ticket
+  relations
+    define project: [project]
+    define owner: [user]
+```
+
+### Wiki
+
+Our wiki model we'll handle in one file for now until it grows some more. Again, we'll also extend the `organization` type here so that we can add a relation to track who can create a space.
+
+```dsl.openfga
+module wiki
+
+extend type organization
+  relations
+    define can_create_space: admin
+
+
+type space
+  relations
+    define organization: [organization]
+    define can_view_pages: member from organization
+
+type page
+  relations
+    define space: [space]
+    define owner: [user]
+```
+
+### `fga.mod`
+
+In order to deploy this model we'll need to create our `fga.mod` manifest file, in here we'll set our schema version and list the individual module files that make up our model.
+
+```yaml
+schema: '1.2'
+contents:
+  - core.fga
+  - issue-tracker/projects.fga
+  - issue-tracker//tickets.fga
+  - wiki.fga
+```
+
+### Putting it all together
+
+Now that we have our individual parts of the modular model, we're able to write this model to <ProductName format={ProductNameFormat.ShortForm}/> and then run tests against it.
+
+In order to write our model we need to use the CLI and run:
+
+```shell
+fga model write --store-id=$FGA_STORE_ID --file fga.mod
+```
+
+We can then write tuples and query this model as you would expect with a singular file authorization model.
+
```

Comment:
Shouldn't we explain what happens when you retrieve the model with `fga model get`?

Reply:
I was thinking we do in a follow up so that we can announce today - but am OK waiting till Tuesday morning to announce.

Reply:
If that goes in the documentation page, we can update it whenever we want.

Reply:
Added a section to the modeling guide that explains the DSL comments as part of 7d25a36be221152d519cdb0ac90ae1f3c0a9feb6